criu/namespaces.c

```diff
@@ -282,6 +282,46 @@ int restore_ns(int rst, struct ns_desc *nd)
 	return ret;
 }
 
+int switch_mnt_ns(int pid, int *rst, int *cwd_fd)
+{
+	int ret;
+	int fd;
+
+	if (!cwd_fd)
+		return switch_ns(pid, &mnt_ns_desc, rst);
+
+	fd = open(".", O_DIRECTORY);
```

Comment:
O_PATH looks more appropriate in this case, doesn't it?

Reply:
Yes, Done~
```
 O_PATH (since Linux 2.6.39)
              Obtain a file descriptor that can be used for two purposes: to indicate a location in the file-system tree and to perform operations that act purely  at  the  file
              descriptor  level
```